src/components/ExtractorResponseEditor/ExtractorResponseEditor.tsx

```diff
@@ -422,16 +422,40 @@ class ExtractorResponseEditor extends React.Component<Props, State> {
         document.removeEventListener("Test_SelectWord", this.onTestSelectWord)
     }
 
-    // For end 2 end unit testing.
+    /**
+     * For cypress end to end unit testing.  Event selects phrase in entity labeller
+     * The following will select the word "hello" in the second row (row is 1 based)
+     *   var event = new CustomEvent("Test_SelectWord", { detail: { phrase: "hello", r:ow: 2 }})
+     *   event.initEvent("Test_SelectWord", true, true)}
+     *   element.dispatchEvent(event)
+     * 
+     * Or default to the first row:
+     *   var event = new CustomEvent("Test_SelectWord", { detail: "hello"})
+     *   event.initEvent("Test_SelectWord", true, true)}
+     *   element.dispatchEvent(event)
+     */
     @autobind
     onTestSelectWord(val: any) {
-        const phrase: string = val.detail
+        if (!val.detail) {
+            throw new Error("Test_SelectWord expecting detail phrase")
+        }
+
+        const phrase: string = val.detail.phrase || val.detail
+        const row: number = val.detail.row - 1 || 0
+
         const words = phrase.split(" ")
         const firstWord = words[0]
         const lastWord = words[words.length - 1]
 
+        // Get row
+        const rows = Array.from(document.querySelectorAll(".entity-labeler"))
```

Comment:
Move to constant and change the selector to `[data-testid="extractor-response-editor-entity-labeler"]`. It's better to avoid code that relies on CSS for presentation when possible.

Reply:
done

Reply:
partially done, but ok